src/Elastic.Transport/Responses/HttpDetails/IApiCallDetails.cs

```diff
@@ -15,23 +15,20 @@ namespace Elastic.Transport
 	/// <summary> Details about the API call.</summary>
 	public interface IApiCallDetails
 	{
-		//TODO: Get rid of setter
         /// <summary>
         /// An audit trail of requests made to nodes within the cluster.
         /// </summary>
-        List<Audit> AuditTrail { get; set; }
+        List<Audit> AuditTrail { get; }
 
-		// TODO: Get rid of setter.
 		/// <summary>
 		/// Thread pool thread statistics collected when making a request.
 		/// </summary>
-		ReadOnlyDictionary<string, ThreadPoolStatistics> ThreadPoolStats { get; set; }
+		ReadOnlyDictionary<string, ThreadPoolStatistics> ThreadPoolStats { get; }
```

Comment:
The setter is currently used elsewhere in the code and can't be removed until that is refactored to be set in `ResponseBuilder.ToResponse`. Can you please leave this in place until that code path is refactored.